src/main/java/internet/market/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,193 @@
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, item.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t add order with id =  " + item.getId(), e);
+        }
+        addProductsToOrder(item.getProducts(), item.getId());
+        return item;
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders WHERE order_id = ? AND deleted = false";
+        Order order = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                order = getOrderInformationFromDataBase(resultSet);
+                order.setProducts(getProductsOfOrder(id));
+                return Optional.of(order);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t find order with id = "
+                    + id, e);
+        }
+        if (order != null) {
+            order.setProducts(getProductsOfOrder(id));
+        }
+        return Optional.ofNullable(order);
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> allOrders = new ArrayList<>();
+        String query = "SELECT * FROM orders WHERE deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                allOrders.add(getOrderInformationFromDataBase(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all orders", e);
+        }
+        for (Order order : allOrders) {
+            order.setProducts(getProductsOfOrder(order.getId()));
+        }
+        return allOrders;
+    }
+
+    @Override
+    public Order update(Order item) {
+        String updateOrderQuery = "UPDATE orders SET user_id = ? "
+                + "WHERE order_id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(updateOrderQuery);
+            statement.setLong(1, item.getUserId());
+            statement.setLong(2, item.getId());
+            statement.executeUpdate();
+            List<Product> allProducts = getProductsOfOrder(item.getId());
+            deleteProductsFromOrder(item.getId());
+            addProductsToOrder(allProducts, item.getId());
```

Comment:
same comment as for shoppingCart `update()` method

Reply:
done